src/lib/eco-router/api.ts

```diff
@@ -173,12 +192,21 @@ export async function getExactOut(
 
   const uniswapV2TradesList = uniswapV2PlatformList.map(async platform => {
     try {
-      const pairs = await getAllCommonPairs({
+      let pairs: Pair[] = []
+
+      const getAllCommonUniswapV2PairsParams = {
         currencyA: currencyAmountOut.currency,
         currencyB: currencyIn,
         platform,
         provider,
-      })
+      }
+
+      if (platform.subgraphEndpoint[chainId] !== undefined) {
+        pairs = await getAllCommonUniswapV2PairsFromSubgraph(getAllCommonUniswapV2PairsParams)
+      } else {
+        pairs = await getAllCommonUniswapV2Pairs(getAllCommonUniswapV2PairsParams)
+      }
```

Comment:
same as the previous comment

Reply:
I feel its fine since this has more readability. Anyway we uglify and minimize our code when we build and deploy.